test/annotations_test.py

```diff
@@ -0,0 +1,45 @@
+import typing
+
+import pytest
+
+import synchronicity
+from synchronicity import Interface
+from synchronicity import async_wrap
+
+
+class _Bar:
+    pass
+
+class _Foo:
+    baz: _Bar
+    async def bar(self, arg1: typing.AsyncIterator[str]):
+        pass
```

Comment:
it would be good to have a test where a vanilla `def` function returns a coroutine, e.g.

```python
async def get_str(self, x):
    return str(x**2)

def baz(self, x: int) -> typing.Coroutine[None, None, str]:
    return get_str(x)
```

Reply:
Good point. Added that now. I did have some annotation mapping tests for nested `Awaitable[...]`s which kind of picks up the same thing, but added `Coroutine` and `Awaitable` to the annotation replacement test now too :+1: